src/vcpkg/paragraphs.cpp

```diff
@@ -433,37 +434,47 @@ namespace vcpkg::Paragraphs
         auto manifest_contents = fs.read_contents(manifest_path, ec);
```

Comment:
I feel like this function should be able to be written with less nesting, perhaps along the lines of:
```cpp
std::error_code ec_manifest, ec_control;
auto content_manifest = fs.read_contents(path_manifest, ec_manifest);
const auto exists_manifest = !ec;
const auto failed_manifest = ec && ec_manifest != std::errc::no_such_file_or_directory;
auto content_control = fs.read_contents(path_control, ec_control);
const auto exists_control = !ec;
const auto failed_control = ec && ec_control != std::errc::no_such_file_or_directory;
if (failed_manifest) {
  // msgFailedToAccessManifest(path_manifest)
} else if (failed_control) {
  // msgFailedToAccessManifest(path_control)
} else if (exists_manifest && exists_control) {
  // msgManifestConflict(port_directory)
} else if (exists_manifest) {
  // success on manifest
} else if (exists_control) {
  // success on control
} else {
  // msgPortMissingManifest2(port_directory)
}
```

Reply:
I don't like this transformation as it does a `read_contents` rather than an exists check on the `CONTROL` in the case we know loading the json succeeded. I inverted the `ec` tests and that flattened the result substantially, does that resolve your hearburn?